metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/data_resolver.py

```diff
@@ -0,0 +1,660 @@
+            POWERBI_BASE_URL=DataResolverBase.BASE_URL,
+            WORKSPACE_ID=workspace.id,
+            REPORT_ID=report_id,
+        )
+        # Hit PowerBi
+        logger.info(f"Request to pages URL={pages_endpoint}")
+        response = self._request_session.get(
+            pages_endpoint,
+            headers={Constant.Authorization: self.get_access_token()},
+        )
+
+        # Check if we got response from PowerBi
+        response.raise_for_status()
+        response_dict = response.json()
+        return [
+            Page(
+                id="{}.{}".format(report_id, raw_instance["name"].replace(" ", "_")),
+                name=raw_instance["name"],
+                displayName=raw_instance.get("displayName"),
+                order=raw_instance.get("order"),
+            )
+            for raw_instance in response_dict["value"]
+        ]
+
+    def get_users(self, workspace_id: str, entity: str, entity_id: str) -> List[User]:
+        return []  # User list is not available in regular access
+
+
+class AdminAPIResolver(DataResolverBase):
+
+    # Admin access endpoints
+    API_ENDPOINTS = {
+        Constant.DASHBOARD_LIST: "{POWERBI_ADMIN_BASE_URL}/groups/{WORKSPACE_ID}/dashboards",
+        Constant.TILE_LIST: "{POWERBI_ADMIN_BASE_URL}/dashboards/{DASHBOARD_ID}/tiles",
+        Constant.REPORT_LIST: "{POWERBI_ADMIN_BASE_URL}/groups/{WORKSPACE_ID}/reports",
+        Constant.SCAN_GET: "{POWERBI_ADMIN_BASE_URL}/workspaces/scanStatus/{SCAN_ID}",
+        Constant.SCAN_RESULT_GET: "{POWERBI_ADMIN_BASE_URL}/workspaces/scanResult/{SCAN_ID}",
+        Constant.SCAN_CREATE: "{POWERBI_ADMIN_BASE_URL}/workspaces/getInfo",
+        Constant.ENTITY_USER_LIST: "{POWERBI_ADMIN_BASE_URL}/{ENTITY}/{ENTITY_ID}/users",
+        Constant.DATASET_LIST: "{POWERBI_ADMIN_BASE_URL}/groups/{WORKSPACE_ID}/datasets",
+    }
+
+    def create_scan_job(self, workspace_id: str) -> str:
+        """
+        Create scan job on PowerBI for the workspace
+        """
+        request_body = {"workspaces": [workspace_id]}
+
+        scan_create_endpoint = AdminAPIResolver.API_ENDPOINTS[Constant.SCAN_CREATE]
+        scan_create_endpoint = scan_create_endpoint.format(
+            POWERBI_ADMIN_BASE_URL=DataResolverBase.ADMIN_BASE_URL
+        )
+
+        logger.debug(
+            f"Creating metadata scan job, request body {request_body}",
+        )
+
+        res = self._request_session.post(
+            scan_create_endpoint,
+            data=request_body,
+            params={
+                "datasetExpressions": True,
+                "datasetSchema": True,
+                "datasourceDetails": True,
+                "getArtifactUsers": True,
+                "lineage": True,
+            },
+            headers={Constant.Authorization: self.get_access_token()},
+        )
+
+        res.raise_for_status()
+        # Return scan_id of Scan created for the given workspace
+        scan_id = res.json()["id"]
```

Comment:
guaranteed to exist?? should we verify??

Reply:
It is guaranteed and first, we are raising an exception just one line above  `res.raise_for_status()` if it is not the 200 response